hr_holidays_leave_repeated/models/hr_leave.py

```diff
@@ -0,0 +1,163 @@
+                "user_error_msg": self.env._(
+                    "The repetition is every four weeks: the duration of the "
+                    "leave request must not exceed 28 days."
+                ),
+            },
+        }
+
+    @api.model
+    def _update_repeated_leave_vals(self, leave, resource_calendar):
+        vals_dict = self._get_repeated_vals_dict()
+        param_dict = vals_dict[leave.repeat_every]
+        from_dt = fields.Datetime.from_string(leave.date_from)
+        to_dt = fields.Datetime.from_string(leave.date_to)
+
+        if (to_dt - from_dt).days > param_dict["days"]:
+            raise UserError(param_dict["user_error_msg"])
+
+        from_dt, to_dt = self._update_repeated_workday_dates(
+            resource_calendar, from_dt, to_dt, param_dict["days"]
+        )
+        client_tz = timezone(self._context.get("tz") or self.env.user.tz or "UTC")
+        request_date_from = utc.localize(from_dt).astimezone(client_tz)
+        request_date_to = utc.localize(to_dt).astimezone(client_tz)
+
+        return {
+            "employee_id": leave.employee_id.id,
+            "date_from": from_dt,
+            "date_to": to_dt,
+            "request_date_from": request_date_from,
+            "request_date_to": request_date_to,
+        }
+
+    @api.model
+    def create_repeated_handler(self, leave, resource_calendar):
+        def _check_repeating(count, leave, date_to):
+            repeat_mode = leave.repeat_mode
+            if repeat_mode == "times" and count < leave.repeat_limit:
+                return True
+            if repeat_mode == "date" and date_to.date() <= leave.repeat_end_date:
+                return True
+            return False
+
+        count = 1
+        leaves = self.env["hr.leave"]
+        vals = self._update_repeated_leave_vals(leave, resource_calendar)
+        while _check_repeating(count, leave, vals.get("date_to")):
+            leave = leave.with_context(skip_create_handler=True).copy(vals)
+            leaves += leave
+            count += 1
+            vals = self._update_repeated_leave_vals(leave, resource_calendar)
+        return leaves
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        skip_create_handler = self.env.context.get("skip_create_handler")
+        if skip_create_handler:
+            return res
+        for leave in res.filtered(
+            lambda leave: leave.repeat_every
+            and leave.repeat_mode
+        ):
+            resource_calendars = leave.employee_id.resource_calendar_id
+            if len(resource_calendars) == 1:
+                res += self.create_repeated_handler(leave, resource_calendars[0])
+            else:
+                raise ValidationError(
+                    self.env._(
+                        "Please define resource calendar on employee(s) in order "
+                        "to compute repeated leaves."
+                    )
+                )
```

Comment:
replace 
```
            resource_calendars = leave.employee_id.resource_calendar_id
            if len(resource_calendars) == 1:
                res += self.create_repeated_handler(leave, resource_calendars[0])
            else:
                raise ValidationError(
                    self.env._(
                        "Please define resource calendar on employee(s) in order "
                        "to compute repeated leaves."
                    )
                )
```
by 
```
res += self.create_repeated_handler(leave, leave.resource_calendar_id)
```
leave.resource_calendar_id is computed field. and its value always be set.